### PR TITLE
CORE-2207 Bookingsync-bootstrap-sass addon - update ember and dependencies

### PR DESCRIPTION
**Pull request by herzzanu**
### References
**Issue link:**
https://bookingsync.atlassian.net/browse/CORE-2207

### CODE REVIEW

File: MODULE_REPORT.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+## Module Report
```
Comment (abel-n) [inline MODULE_REPORT.md]: This is a log file.

File: addon/components/bsy-button.js [hunk 2]
```diff
@@ -4,31 +4,32 @@ import layout from '../templates/components/bsy-button';
...
+  classNames: 'btn',
+  classNameBindings: ['typeClassName', 'loadingClassName'],
   isLoading: false,
-  loadingLabel: "Loading...",
+  loadingLabel: 'Loading...',
 
   typeClassName: computed('type', function() {
-    if (this.get('type') === "submit") {
-      return "btn-primary";
+    if (this.type === 'submit') {
```
Comment (abel-n) [inline addon/components/bsy-button.js]: This and the CP below could return a ternary conditional or the `else` branch is unnecessary.

File: addon/components/bsy-button.js [hunk 3]
```diff
@@ -4,31 +4,32 @@ import layout from '../templates/components/bsy-button';
...
     }
   }),
 
-  click: function() {
-    this.set("isLoading", true);
-    this.$().prop('disabled', 'disabled');
-    this.$().text(this.get("loadingLabel"));
+  click() {
+    this.set('isLoading', true);
+    this.element.disabled = true;
```
Comment (abel-n) [inline addon/components/bsy-button.js]: This could be handled by `attributeBindings`.

File: addon/components/bsy-button.js [hunk 4]
```diff
@@ -4,31 +4,32 @@ import layout from '../templates/components/bsy-button';
...
+    this.element.textContent = this.loadingLabel;
```
Comment (abel-n) [inline addon/components/bsy-button.js]: This could be handled by template.

File: addon/components/bsy-button/close.js [hunk 5]
```diff
@@ -3,10 +3,11 @@ import layout from '../../templates/components/bsy-button/close';
...
-  tagName: "button",
-  classNames: "btn btn-secondary-inverse btn-round",
 
-  click: function() {
-    this.$().prop('disabled', 'disabled');
+  tagName: 'button',
+  classNames: 'btn btn-secondary-inverse btn-round',
+
+  click() {
+    this.element.disabled = true;
```
Comment (abel-n) [inline addon/components/bsy-button/close.js]: This could be handled by `attributeBindings`.

File: addon/components/bsy-input.js [hunk 6]
```diff
@@ -1,42 +1,43 @@
...
-  inputClassNames: computed('size', function () {
+  inputClassNames: computed('size', function() {
     let size = this.get('size');
 
     if (size) {
       return `form-control input-${size}`;
     }
 
-    return "form-control";
+    return 'form-control';
```
Comment (abel-n) [inline addon/components/bsy-input.js]: This can be a ternary conditional.

File: addon/components/bsy-input.js [hunk 7]
```diff
@@ -1,42 +1,43 @@
...
   }),
 
-  iconSize: computed('size', function () {
-    let size = this.get('size');
-
-    if (size) {
-      return `icon--${size}`;
-    }
+  iconSize: computed('size', function() {
+    let { size } = this;
```
Comment (abel-n) [inline addon/components/bsy-input.js]: I believe in these cases we don't define a variable but just refer to it via `this`. Same in `formGroupSize`.

File: addon/components/bsy-nav/item.js [hunk 8]
```diff
@@ -6,14 +6,14 @@ export default Component.extend({
   layout,
   tagName: 'li',
   classNames: 'nav-item',
-  classNameBindings: ["activeClassName"],
+  classNameBindings: ['activeClassName'],
   active: false,
 
   activeClassName: computed('active', function() {
```
Comment (abel-n) [inline addon/components/bsy-nav/item.js]: Instead of this whole CP we can just put `active` into `classNameBindings` if I get it right.

Comment (abel-n) [reply addon/components/bsy-nav/item.js]: Same in `addon/components/bsy-nav/link.js`.

File: addon/components/bsy-radio-buttons/button.js [hunk 9]
```diff
@@ -11,7 +11,7 @@ const BsyRadioButtonsButtonComponent = Component.extend({
 
   activeClassName: computed('value', function() {
     return isEqual(this.get('value'), this.get('option'));
```
Comment (abel-n) [inline addon/components/bsy-radio-buttons/button.js]: This is wrong in several ways. I believe we don't need to `.get()` and a dependent key is probably missing.

File: addon/components/bsy-button/close.js (hunk 5 above)
Comment (herzzanu) [reply addon/components/bsy-button/close.js]: Yes, but in this case we need to introduce another attribute.

File: addon/components/bsy-input.js (hunk 7 above)
Comment (herzzanu) [reply addon/components/bsy-input.js]: Not a rule but sure let's have it like that.

File: addon/components/bsy-nav/item.js (hunk 8 above)
Comment (herzzanu) [reply addon/components/bsy-nav/item.js]: Nice catch

File: addon/components/bsy-button/close.js (hunk 5 above)
Comment (abel-n) [reply addon/components/bsy-button/close.js]: We can have it as is, but if I'd written the code I would've gone the Ember way. :+1:

Comment (herzzanu) [reply addon/components/bsy-button/close.js]: Agree, still room for a lot of improvement. This addon (the ember part) will eventually be faded out though.

**Pull request merged**